guacamole/src/main/frontend/src/app/client/controllers/clientController.js

```diff
@@ -485,17 +498,23 @@ angular.module('client').controller('clientController', ['$scope', '$routeParams
         $scope.menu.connectionParameters = newFocusedClient ?
             ManagedClient.getArgumentModel(newFocusedClient) : {};
 
+        // Re-broadcast the updated client
+        $scope.$broadcast('guacClientChanged'. newFocusedClient);
+
     });
 
     // Automatically update connection parameters that have been modified
     // for the current focused client
     $scope.$on('guacClientArgumentsUpdated', function focusedClientChanged(event, focusedClient) {
 
-        // Update available connection parameters, if the updated arguments are
-        // for the current focused client - otherwise ignore them
-        if ($scope.focusedClient && $scope.focusedClient === focusedClient)
+        // Ignore any updated arguments not for the current focused client
+        if ($scope.focusedClient && $scope.focusedClient === focusedClient) {
             $scope.menu.connectionParameters = ManagedClient.getArgumentModel(focusedClient);
 
+            // Re-broadcast the updated arguments
+            $scope.$broadcast('guacClientArgumentsChanged'. focusedClient);
```

Comment:
Is there a way that `guacClientArgumentsUpdated` and `guacClientArgumentsChanged` might be renamed to avoid confusing the two?

Reply:
Hmm, I do agree that they're confusingly similar names (seeing as they're basically the same thing, just going in opposite directions), but I'm not sure what the right name would be... 

I'd like to keep `guacClientArgumentsChanged` since it follows the same format as the existing `guacClientChanged` event.

Did you have any ideas?

Reply:
Eh, I guess these events are really so similar they could probably just have the same names. Any consumers of the events won't care whether they're going up or down the stack. I'll change them to have the same names.

Reply:
Ok - new approach - we can just broadcast all these events directly from the root scope, which obviates these duplicated event names entirely. Any consumer can just `$scope.$on()` them as before. How does this look?

Reply:
Makes sense - this looks much better.